barman/backup.py

```diff
@@ -697,6 +701,23 @@ def backup(self, wait=False, wait_timeout=None, name=None, **kwargs):
 
         finally:
             if backup_info:
+
+                # IF is an incremental backup, we save here child backup info id
+                # inside the parent list of children. no matter if the backup
+                # is successful or not. This is needed to be able to retrieve
+                # also failed incremental backups for removal or other operations
+                # like show-backup.
+                parent_backup_info = backup_info.get_parent_backup_info()
+
+                if parent_backup_info:
```

Comment:
I did some thinking here and dumped most of it in the above comment. i think that doing this save here is the way to go @barthisrael

Reply:
That makes sense!
Can you believe I had a similar through while I was driving -- but because of that I was not able to tell you 😆 
I'm happy with that!